### PR TITLE
SceneObjectBase: Fixes issue with useState subscription misses state change that happens between frist render and useEffect 

### PR DESCRIPTION
**Pull request by torkelo**

### CODE REVIEW

File: packages/scenes/src/core/SceneObjectBase.tsx [hunk 1]
```diff
@@ -278,12 +277,19 @@ export abstract class SceneObjectBase<TState extends SceneObjectState = SceneObj
...
 function useSceneObjectState<TState extends SceneObjectState>(model: SceneObjectBase<TState>): TState {
-  const forceUpdate = useForceUpdate();
+  const [_, setState] = useState<TState>(model.state);
+  const stateAtFirstRender = model.state;
 
   useEffect(() => {
-    const s = model.subscribeToState(forceUpdate);
+    const s = model.subscribeToState(setState);
+
+    if (model.state !== stateAtFirstRender) {
```
Comment (dprokop) [inline packages/scenes/src/core/SceneObjectBase.tsx:286]:
```suggestion
    // Re-render component if the state changed before state subscription (model.subscribeToState) has been called.
    if (model.state !== stateAtFirstRender) {
```

**Review by dprokop — APPROVED** [review]

File: packages/scenes/src/core/SceneObjectBase.tsx (hunk 1 above)
Comment (dprokop) [reply packages/scenes/src/core/SceneObjectBase.tsx:286]: Wait this comment is not 100% true:D

Comment (dprokop) [reply packages/scenes/src/core/SceneObjectBase.tsx:286]: Should be better now

Comment (torkelo) [reply packages/scenes/src/core/SceneObjectBase.tsx:287]: @dprokop what about this comment 
// Re-render component if the state changed between first render and useEffect (mount)

Comment (dprokop) [reply packages/scenes/src/core/SceneObjectBase.tsx:287]: Great

**Pull request merged**